src/librustc_mir/transform/simplify_try.rs

```diff
@@ -306,7 +306,11 @@ fn optimization_applies<'tcx>(
 }
 
 impl<'tcx> MirPass<'tcx> for SimplifyArmIdentity {
-    fn run_pass(&self, _: TyCtxt<'tcx>, source: MirSource<'tcx>, body: &mut Body<'tcx>) {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, source: MirSource<'tcx>, body: &mut Body<'tcx>) {
+        if tcx.sess.opts.debugging_opts.mir_opt_level < 2 {
```

Comment:
Uh, could we maybe not keep buggy passes running at all? I think a `-Zmir-opt-level=3` miscompilation is a soundness bug just like it would be at a lower optimization level.

C/C++ programmers seem to associate "high optimization levels" with "creates buggy code", but really that's awful and we shouldn't do it.

Reply:
@RalfJung I opened a related MCP: https://github.com/rust-lang/compiler-team/issues/319

If you have thoughts about it, please leave a comment there. Thanks!